### ISSUE TITLE
Allow for custom variables to be passed via the cli

### ISSUE THREAD

**Issue by pecostef** [serverless-stack/sst#2228]
Hello!
First of all thanks for the amazing work, I do really love to work with SST!
Related to the app I'm developing with SST,  I have some CDK code that needs to be executed in order to add permission sets to AWS Identity Center that is managed in a different account (I am using Cognito with SAML auth backed up by Identity Center).
In order to conditionally deploy the app or the SSO Infra stacks, I wanted to pass a parameter to the index.ts entrypoint; something like the following: `npx sst deploy --deploy-sso-infra`
However I cannot find the variable neither in `process.argv` nor in `process.env`.
I know I could use a made-up stage name (e.g. something like `npx sst deploy --stage sso `) and and that's actually what I am doing at the moment, but I would like to be more explicit and I generally think it could be useful also if you want define stakcs with parameters ( ` new CfnParameter(...) )`.

Thanks again for the awesom work  :)

**Comment by fwang** [comment]
Hey @pecostef, can u use environment variable instead?

ie. `DEPLOY_SSO_INFRA=true npx sst deploy` and check for `process.env. DEPLOY_SSO_INFRA`

**Comment by pecostef** [comment]
Hi @fwang ! Thanks for your comment.
Sure I can use env. variables. I just noticed that the --parameters flag that you would normally use with the CDK are not passed over by sst.
However, thinking a bit more about this, I also kind of realized that the philosophy behind sst is a bit different, in the sense that it is meant to develop and deploy self-contained apps and in this context having parametrized stacks to handle different types of infra is IMHO an anti-pattern.
What I ended up doing is extracting the cross-apps infrastructure into a separate CDK project, so that shared infra needed for all the different apps like SSO, CI/CD, etc are handled separately.

Once more thanks for the amazing work, SST is awesome!
